internal/mode/static/nginx/config/servers.go

```diff
@@ -544,8 +543,8 @@ func createPath(rule dataplane.PathRule) string {
 	}
 }
 
-func createPathForMatch(path string, pathType dataplane.PathType, routeIdx int) string {
-	return fmt.Sprintf("%s_%s_route%d", path, pathType, routeIdx)
+func createPathForMatch(routeIdx int) string {
+	return fmt.Sprintf("/%s_rule%d_route%d", dataplane.PathTypeInternal, routeIdx, routeIdx)
```

Comment:
`dataplane.PathTypeInternal` has a `/` prefix, so I'm confused how we aren't getting double slashes here.

Also, I think there can be overlap in `routeIdx` between rules, because this index is inside a nested for loop. In `createLocations`, this index is regenerated for every top level loop iteration.

Reply:
will update